napari/utils/_tests/test_translations.py

```diff
@@ -0,0 +1,177 @@
+    distinfo = tmp_path / "napari_language_pack_es_CO-0.1.0.dist-info"
+    distinfo.mkdir()
+    (distinfo / "top_level.txt").write_text('napari_language_pack_es_CO')
+    (distinfo / "entry_points.txt").write_text(
+        "[napari.languagepack]\nes_CO = napari_language_pack_es_CO\n"
+    )
+    (distinfo / "METADATA").write_text(
+        "Metadata-Version: 2.1\n"
+        "Name: napari-language-pack-es-CO\n"
+        "Version: 0.1.0\n"
+    )
+    pkgdir = tmp_path / 'napari_language_pack_es_CO'
+    msgs = pkgdir / 'locale' / 'es_CO' / 'LC_MESSAGES'
+    msgs.mkdir(parents=True)
+    (pkgdir / '__init__.py').touch()
+    (msgs / "napari.po").write_text(es_CO_po)
+    (msgs / "napari.mo").write_bytes(es_CO_mo)
+
+    from napari_plugin_engine.manager import temp_path_additions
+
+    with temp_path_additions(tmp_path):
+        # Load translator and force a locale for testing
+        translator._set_locale(TEST_LOCALE)
+        return translator.load()
+
+
+def test_get_language_packs(trans):
+    result = get_language_packs()
+    assert result == {
+        'en': {'displayName': 'English', 'nativeName': 'English'}
+    }
+
+
+def test_get_display_name_valid():
+    assert _get_display_name("en", "en") == "English"
+    assert _get_display_name("en", "es") == "Inglés"
+    assert _get_display_name("en", "es_CO") == "Inglés"
+    assert _get_display_name("en", "fr") == "Anglais"
+    assert _get_display_name("es", "en") == "Spanish"
+    assert _get_display_name("fr", "en") == "French"
+
+
+def test_get_display_name_invalid():
+    assert _get_display_name("en", "foo") == "English"
+    assert _get_display_name("foo", "en") == "English"
+    assert _get_display_name("foo", "bar") == "English"
+
+
+def test_is_valid_locale_valid():
+    assert _is_valid_locale("en")
+    assert _is_valid_locale("es")
+    assert _is_valid_locale("es_CO")
+
+
+def test_is_valid_locale_invalid():
+    assert not _is_valid_locale("foo_SPAM")
+    assert not _is_valid_locale("bar")
+
+
+def test_locale_valid_singular(trans):
+    # Test singular method
+    expected_result = "Más sobre napari"
+    result = trans.gettext("MORE ABOUT NAPARI")
+    assert result == expected_result
+
+    # Test singular method shorthand
+    result = trans._("MORE ABOUT NAPARI")
+    assert result == expected_result
+
+
+def test_locale_invalid():
+    with pytest.warns(UserWarning):
+        translator._set_locale(TEST_LOCALE)
+        trans = translator.load()
+        result = trans._("BOO")
+        assert result == "BOO"
+
+
+def test_locale_n_runs(trans):
+    # Test plural method
+    n = 2
+    string = "MORE ABOUT NAPARI"
+    plural = "MORE ABOUT NAPARIS"
+    result = trans.ngettext(string, plural, n)
+    assert result == plural
+
+    # Test plural method shorthand
+    result = trans._n(string, plural, n)
+    assert result == plural
+
+
+@pytest.mark.xfail(
+    reason="On ubuntu and python3.7 is failing, seems to be CI only"
```

Comment:
can you point me to the failure on CI?
If it's just ubuntu and 3.7, can we skip only that rather than xfailing everything?

Reply:
Fixed @tlambert03 !